### PR TITLE
magento/magento2#26749: Product flat table - Incorrect data

### LINKED ISSUE #26749: Product flat table - Incorrect data
We're current facing an issue that the product data has been inserted to an incorrect flat table. 

### Preconditions (*)
1. Magento version 2.3.3 with Sample Data
2. Enable Catalog Product Flat.
3. Index mode for Catalog Product Flat: Update on save

### Steps to reproduce (*)
1. Create 2 new websites, for each website, create one store view (let say the store view id for website 1 is 2, store view id for website 2 is 3).
2. Come to any product detail, try to assign this product to website 1, then click Save.

### Expected result (*)
1. Product information just insert to flat table of the website that product is assigned to (in this case, this table is catalog_product_flat_2)

### Actual result (*)
1. Product information has been inserted for all product flat table (in this case, include catalog_product_flat_3)

After i run `php bin/magento indexer:reindex catalog_product_flat` all invalid data has been removed (in this case, this product information has been removed in catalog_product_flat_3 ).
Please help to check it again.
Thanks.

### PR DESCRIPTION
**Pull request by engcom-Charlie**
### Description (*)

Fixed product data has been inserted into an incorrect flat table. Covered by test.
### Fixed Issues (if relevant)

1. Fixes magento/magento2#26749: Product flat table - Incorrect data

### Manual testing scenarios (*)

Please see https://github.com/magento/magento2/issues/26749

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Review by dmytro-ch — CHANGES_REQUESTED** [review]
Hello @engcom-Charlie! Could you please also update the unit test?
Thank you!

**Comment by engcom-Charlie** [comment]
> Hello @engcom-Charlie! Could you please also update the unit test?
> Thank you!

Hello @dmytro-ch, yes I'll do it.

**Comment by engcom-Charlie** [comment]
@magento run Functional Tests B2B

**Comment by engcom-Charlie** [comment]
Hello @ihor-sviziev, could You please make a review on this PR?

File: app/code/Magento/Catalog/Model/Indexer/Product/Flat/Action/Row.php [hunk 1]
```diff
@@ -64,36 +92,36 @@ public function __construct(
...
+     * @return Row
+     * @throws LocalizedException
      */
     public function execute($id = null)
     {
-        if (!isset($id) || empty($id)) {
-            throw new \Magento\Framework\Exception\LocalizedException(
-                __('We can\'t rebuild the index for an undefined product.')
-            );
+        if (!isset($id) || $id === null) {
```
Comment (dmytro-ch) [inline app/code/Magento/Catalog/Model/Indexer/Product/Flat/Action/Row.php]: The `$id === null` condition is redundant in this case isn't it?

**Review by dmytro-ch — COMMENTED** [review]
Hello @engcom-Charlie, sorry for the late reply. I've missed the review request notification for some reason.
The result looks good to me.
Could you please just check the minor CR notice?

Thank you!

**Review by dmytro-ch — DISMISSED** [review]

**Comment by magento-engcom-team** [comment]
Hi @dmytro-ch, thank you for the review.
[ENGCOM-7211](https://jira.corp.magento.com/browse/ENGCOM-7211) has been created to process this Pull Request

**Comment by engcom-Echo** [comment]
Hi, @engcom-Charlie will resolve conflict.

**Review by engcom-Alfa — CHANGES_REQUESTED** [review]
Hi @engcom-Charlie. Could you resolve merge conflicts?
Thanks!

**Comment by engcom-Echo** [comment]
Close PR because this issue has already been fixed: https://github.com/magento/magento2/pull/27365

**Pull request closed without merging**